src/modules/dashboard/sagas/extensions.ts

```diff
@@ -0,0 +1,244 @@
+  yield apolloClient.query<ColonyExtensionQuery, ColonyExtensionQueryVariables>(
+    {
+      query: ColonyExtensionDocument,
+      variables: {
+        colonyAddress,
+        extensionId,
+      },
+      fetchPolicy: 'network-only',
+    },
+  );
+}
+
+function* colonyExtensionInstall({
+  meta,
+  payload: { colonyAddress, extensionId },
+}: Action<ActionTypes.COLONY_EXTENSION_INSTALL>) {
+  const txChannel = yield call(getTxChannel, meta.id);
+
+  try {
+    yield fork(createTransaction, meta.id, {
+      context: ClientType.ColonyClient,
+      methodName: 'installExtension',
+      identifier: colonyAddress,
+      params: [getExtensionHash(extensionId), 1],
+    });
+
+    yield takeFrom(txChannel, ActionTypes.TRANSACTION_CREATED);
+
+    yield put<AllActions>({
+      type: ActionTypes.COLONY_EXTENSION_INSTALL_SUCCESS,
+      payload: {},
+      meta,
+    });
+
+    yield takeFrom(txChannel, ActionTypes.TRANSACTION_SUCCEEDED);
+
+    yield call(refreshExtension, colonyAddress, extensionId);
+  } catch (error) {
+    return yield putError(
+      ActionTypes.COLONY_EXTENSION_INSTALL_ERROR,
+      error,
+      meta,
+    );
+  } finally {
+    txChannel.close();
+  }
+  return null;
+}
+
+function* colonyExtensionEnable({
+  meta,
+  payload: { colonyAddress, extensionId, ...payload },
+}: Action<ActionTypes.COLONY_EXTENSION_ENABLE>) {
+  const extension = extensionData[extensionId];
+  const initChannelName = `${meta.id}-initialise`;
+  const setPermissionChannelName = `${meta.id}-setUserRoles`;
+
+  if (!extension) {
+    throw new Error(`Extension with id ${extensionId} does not exist!`);
+  }
+  const initChannel = yield call(getTxChannel, initChannelName);
+  const setPermissionChannel = yield call(
+    getTxChannel,
+    setPermissionChannelName,
+  );
+  const apolloClient = TEMP_getContext(ContextModule.ApolloClient);
+
+  try {
+    const { data } = yield apolloClient.query<
+      ColonyExtensionQuery,
+      ColonyExtensionQueryVariables
+    >({
+      query: ColonyExtensionDocument,
+      variables: {
+        colonyAddress,
+        extensionId,
+      },
+    });
+
+    if (!data) {
+      throw new Error('Extension not installed');
+    }
+
+    const {
+      // address,
+      details: { initialized, missingPermissions },
+    } = data.colonyExtension;
+
+    if (!initialized && extension.initializationParams) {
+      const initParams = extension.initializationParams.map(
+        ({ paramName }) => payload[paramName],
+      );
+      yield fork(createTransaction, initChannelName, {
+        context: `${extensionId}Client`,
+        methodName: 'initialise',
+        identifier: colonyAddress,
+        params: initParams,
+        group: {
+          key: 'enableExtension',
+          id: meta.id,
+          index: 0,
+        },
+      });
+    }
+
+    if (missingPermissions.length) {
+      // @TODO get the bytes32roles somehow
+      // yield fork(createTransaction, setPermissionChannelName, {
```

Comment:
There's a bunch more left-overs it this file, but since I assume this is not yet done, I guess it makes sense :)

Reply:
Fixed now